### ISSUE TITLE
Add a category to the icons yml file for each icon

### ISSUE THREAD

**Issue by chrisconnors-ibm** [carbon-design-system/carbon#2672]
edit https://github.com/carbon-design-system/carbon/blob/master/packages/icons/metadata.yml#L5

to add the categories reflected here: https://www.ibm.com/design/language/elements/icon-library/

Action
Controls
Data
File
Formatting
Image
Navigation
Social
Status
Toggle
User

For each icon

**Comment by chrisconnors-ibm** [comment]
@jeanservaas @aagonzales @shixiedesign @conradennis @designertyler @laurenmrice this is the issue to add category values to the yaml file. @joshblack 's plugin will depend on this value to organize them into 11 menus.

**Comment by joshblack** [comment]
Here's a quick how-to from @shixiedesign that she wrote up for editing the file: https://github.com/carbon-design-system/carbon/blob/master/docs/icons.md#updating-icon-metadata

Should we be updating `categories` from a list to just `category`? https://github.com/carbon-design-system/carbon/blob/master/packages/icons/metadata.yml#L5

**Comment by shixiedesign** [comment]
I think categories list still make sense. @joshblack An edit pencil may be in the **Image** category and also **Tools**. A few potential issue with categories listed above: 

- a lot of icons currently not in sketch kit are not given a category yet
- image category has over 100~ icons while everything else have only 10-20. Image is currently synonymous to "misc" 
- this is erica's 1 quick pass and she has always intended to re-do it with better categorization

So I believe we should make this a work item like this: 

1. identifying the best categories
2. do a team activity to sort the icons.

**Comment by chrisconnors-ibm** [comment]
We need a category value for display categorization (single value), and synonyms to broaden search results (multiple values). 

Categories are going to drive the menu structure in the Sketch library, so we can’t have icons in multiple “categories”.

**Comment by joshblack** [comment]
Yeah, could get tricky with multiple categories. It seems like it'd be nice to be able to have different ways to find icons though, especially on the website. If we want to, we could generate each asset per category. I feel like the consequence of that will be if people use the same icon from multiple categories it could get confusing in a single sketch file 🤔

**Comment by shixiedesign** [comment]
That make sense @chrisconnors-ibm. I take it back! Think it's a good idea to roll with 1 category for each icon, at least for now since we are restricted by Sketch's folder system.

**Comment by chrisconnors-ibm** [comment]
We’re already set up in the .yml for a single unique `category` value and multiple `alias` values, so it should be fine.

**Comment by jeanservaas** [comment]
Is someone doing this work, or are we gonna break it up like last time?

**Comment by shixiedesign** [comment]
@chrisconnors-ibm No actually. the .yml is set up for multiple category entries. You should look at 
https://github.com/carbon-design-system/carbon/blob/master/packages/icons/metadata.yml
```
categories:
      - example cateogry
```

@joshblack does this mean you will have to do a pass on the file to batch change the format of this field?

**Comment by joshblack** [comment]
@shixiedesign if we want we can keep the list format but say we only keep one entry in it? That way we have the option in the future of having multiple ones

**Comment by chrisconnors-ibm** [comment]
I added categories up to L1394 `warning` and committed to my branch

**Comment by chrisconnors-ibm** [comment]
```  - name: zoom
    friendly_name: Zoom
    usage: This is a description for usage
    categories:
      - Navigation
    aliases:
      - expand
      - inspect
      - size
      - enlarge
      - detail
    variants:
      - name: zoom--in
        friendly_name: Zoom in
        usage: This is a description for usage
        sizes:
          - 16
          - 32
      - name: zoom--out
        friendly_name: Zoom out
        usage: This is a description for usage
        sizes:
          - 16
          - 32
```

It may only be by convention, but I have't any with more than one category, and many with multiple aliases. I'm up to L1455

**Comment by chrisconnors-ibm** [comment]
@jeanservaas I've been doing them since Friday but could certainly use some help. I'd say we split it into thirds, so 
L1-L2280 - me
L2281-L4558 - Available
L4559-L6838 - Available

I'm using the categories here: https://www.ibm.com/design/language/elements/icon-library/

(even though I know they are going to change - @conradennis is developing a new schema

**Comment by chrisconnors-ibm** [comment]
I replaced every instance of "example Category" with "TBD" so that if we build kits before all the categories are settled, there will be some under a "TBD" menu rather than Example Category.

**Comment by joshblack** [comment]
Closed by #2934
